tests/installation/test_executor.py

```diff
@@ -739,6 +776,42 @@ def test_executor_should_write_pep610_url_references_for_urls(
     verify_installed_distribution(tmp_venv, package, expected_url_reference)
 
 
+def test_executor_should_not_write_pep610_url_references_for_non_wheel_urls(
+    tmp_venv: VirtualEnv,
+    pool: RepositoryPool,
+    config: Config,
+    io: BufferedIO,
+    mock_file_downloads: None,
+):
```

Comment:
You can verify that the hash is not written if a wheel is cached by extending your test case as follows:

(The mock has to be adapted after implementing the `strict` behavior.)

```suggestion
@pytest.mark.parametrize("cached", [False, True])
def test_executor_should_write_pep610_url_references_for_non_wheel_urls(
    tmp_venv: VirtualEnv,
    pool: RepositoryPool,
    config: Config,
    io: BufferedIO,
    mock_file_downloads: None,
    mocker: MockerFixture,
    fixture_dir: FixtureDirGetter,
    cached: bool,
):
    if cached:
        link_cached = fixture_dir("distributions") / "demo-0.1.0-py2.py3-none-any.whl"
        mocker.patch(
            "poetry.installation.chef.Chef.get_cached_archive_for_link",
            return_value=link_cached,
        )
```

Reply:
Done